src/provider/utils.ts

```diff
@@ -31,11 +31,33 @@ export function txIdentifier(txHash?: BigNumberish, txId?: BigNumberish): string
 // hex string and BN are detected as block hashes
 // decimal string and number are detected as block numbers
 // null appends nothing to the request url
+
 export type BlockIdentifier = BlockNumber | BigNumberish;
 type BlockIdentifierObject =
   | { type: 'BLOCK_NUMBER'; data: BlockNumber }
   | { type: 'BLOCK_HASH'; data: BigNumberish };
 
+
+export class BlockIdentifierClass {
```

Comment:
Come to think of it, couldn't we have just used the existing getBlockIdentifier function from utils? Maybe would need to be edited a bit, BLOCK_HASH to lower case etc